chapters/interface.tex

```diff
@@ -598,32 +607,23 @@ \section{Type Compatible Expressions}\label{type-compatible-expressions}
 \item
-  If \lstinline!A! is a scalar expression of a simple type \lstinline!B! must also be a scalar
-  expression of a simple type.
+  If \lstinline!A! is a scalar expression of a simple type \lstinline!B! must also be a scalar expression of a simple type.
 \item
-  If \lstinline!A! is a \lstinline!Real! expression then \lstinline!B! must be a \lstinline!Real! or \lstinline!Integer! expression
-  and the type compatible expression is \lstinline!Real!, compare \cref{standard-type-coercion}.
+  If \lstinline!A! is a \lstinline!Real! expression then \lstinline!B! must be a \lstinline!Real! or \lstinline!Integer! expression and the type compatible expression is \lstinline!Real!, compare \cref{standard-type-coercion}.
 \item
   If \lstinline!A! is an \lstinline!Integer! expression then \lstinline!B! must be a \lstinline!Real! or \lstinline!Integer! expression.
   For exponentiation and division the type compatible expression is \lstinline!Real! (even if both \lstinline!A! and \lstinline!B! are \lstinline!Integer!) see \cref{element-wise-exponentiation} and \cref{division-by-numeric-scalars}, in other cases the type compatible expression is \lstinline!Real! or \lstinline!Integer! (same as \lstinline!B!), compare \cref{standard-type-coercion}.
 \item
-  If \lstinline!A! is a \lstinline!Boolean! expression then \lstinline!B! must be a \lstinline!Boolean! expression and
-  the type compatible expression is \lstinline!Boolean!.
+  If \lstinline!A! is a \lstinline!Boolean! expression then \lstinline!B! must be a \lstinline!Boolean! expression and the type compatible expression is \lstinline!Boolean!.
 \item
-  If \lstinline!A! is a \lstinline!String! expression then \lstinline!B! must be a \lstinline!String! expression and the
-  type compatible expression is \lstinline!String!.
+  If \lstinline!A! is a \lstinline!String! expression then \lstinline!B! must be a \lstinline!String! expression and the type compatible expression is \lstinline!String!.
 \item
-  If \lstinline!A! is an enumeration expression then \lstinline!B! must be an enumeration
-  expression and the type compatible expression is enumeration
-  expression, and all enumeration expressions must be defined in terms
-  of an enumeration type with the same enumeration literals in the same
-  order.
+  If \lstinline!A! is an enumeration expression then \lstinline!B! must be an enumeration expression and the type compatible expression is enumeration expression.
+  The enumeration expressions must be defined in terms of an enumeration type with the same enumeration literals in the same order.
 \item
   If \lstinline!A! has an \lstinline!operator record! base class then \lstinline!B! must also have an \lstinline!operator record! base class, and it must be the same, and otherwise neither \lstinline!A! nor \lstinline!B! may have an \lstinline!operator record! base class.
```

Comment:
That's not entirely true, is it? For binary operators, the statement on line 595 of this PR applies. Hmmm, at the top of this itemized list it says that it's about "the resulting type of the expression in case of two type compatible subexpressions". So, it technically already excludes the case when A and B are not type compatible. However, in reality, this list also gives the explanation of when A and B are type compatible. Even this sentence it states the requirements for B, not just what the result would be. So maybe it is best to also change the sentence that introduces this list.

Reply:
Ok, makes sense.